alerting/src/main/kotlin/org/opensearch/alerting/DocumentLevelMonitorRunner.kt

```diff
@@ -703,429 +496,45 @@ class DocumentLevelMonitorRunner : MonitorRunner() {
-                        "Error: ${e.message}",
-                    e
-                )
-                if (e is IndexClosedException) {
-                    throw e
-                }
-            }
-            if (
-                transformedDocs.isNotEmpty() &&
-                shouldPerformPercolateQueryAndFlushInMemoryDocs(transformedDocs.size, monitorCtx)
-            ) {
-                performPercolateQueryAndResetCounters(
-                    monitorCtx,
-                    monitor,
-                    monitorMetadata,
-                    monitorInputIndices,
-                    concreteIndices,
-                    inputRunResults,
-                    docsToQueries,
-                )
-            }
-        }
-    }
-
-    private fun shouldPerformPercolateQueryAndFlushInMemoryDocs(
-        numDocs: Int,
-        monitorCtx: MonitorRunnerExecutionContext,
-    ): Boolean {
-        return isInMemoryDocsSizeExceedingMemoryLimit(docsSizeOfBatchInBytes, monitorCtx) ||
-            isInMemoryNumDocsExceedingMaxDocsPerPercolateQueryLimit(numDocs, monitorCtx)
-    }
-
-    private suspend fun performPercolateQueryAndResetCounters(
-        monitorCtx: MonitorRunnerExecutionContext,
-        monitor: Monitor,
-        monitorMetadata: MonitorMetadata,
-        monitorInputIndices: List<String>,
-        concreteIndices: List<String>,
-        inputRunResults: MutableMap<String, MutableSet<String>>,
-        docsToQueries: MutableMap<String, MutableList<String>>,
-    ) {
-        try {
-            val percolateQueryResponseHits = runPercolateQueryOnTransformedDocs(
-                monitorCtx,
-                transformedDocs,
-                monitor,
-                monitorMetadata,
-                concreteIndices,
-                monitorInputIndices,
-            )
-
-            percolateQueryResponseHits.forEach { hit ->
-                var id = hit.id
-                concreteIndices.forEach { id = id.replace("_${it}_${monitor.id}", "") }
-                monitorInputIndices.forEach { id = id.replace("_${it}_${monitor.id}", "") }
-                val docIndices = hit.field("_percolator_document_slot").values.map { it.toString().toInt() }
-                docIndices.forEach { idx ->
-                    val docIndex = "${transformedDocs[idx].first}|${transformedDocs[idx].second.concreteIndexName}"
-                    inputRunResults.getOrPut(id) { mutableSetOf() }.add(docIndex)
-                    docsToQueries.getOrPut(docIndex) { mutableListOf() }.add(id)
-                }
-            }
-            totalDocsQueriedStat += transformedDocs.size.toLong()
-        } finally {
-            transformedDocs.clear()
-            docsSizeOfBatchInBytes = 0
-        }
+    private fun getNodes(monitorCtx: MonitorRunnerExecutionContext): Map<String, DiscoveryNode> {
+        return monitorCtx.clusterService!!.state().nodes.dataNodes.filter { it.value.version >= Version.CURRENT }
```

Comment:
If someone simply upgrades from old 2.13 RC to new 2.13 RC, would this work?

Reply:
even if this action is not found in remote node, it is retried on local node.https://github.com/opensearch-project/alerting/blob/49b2374cfe3123a0e0ce87bf6813e033f568dee2/alerting/src/main/kotlin/org/opensearch/alerting/DocumentLevelMonitorRunner.kt#L286